crates/object-store-util/src/lib.rs

```diff
@@ -149,6 +145,64 @@ pub async fn create_object_store_client(
     Ok(object_store)
 }
 
+/// Convert Restate [RetryPolicy] into [object_store::RetryConfig].
+fn from_retry_policy(retry_policy: &RetryPolicy) -> RetryConfig {
+    match retry_policy {
+        RetryPolicy::None => RetryConfig {
+            max_retries: 0, // zero disables retries in object_store::RetryConfig
+            ..Default::default()
+        },
+        RetryPolicy::FixedDelay {
+            interval,
+            max_attempts,
+        } => RetryConfig {
+            // retry policy with unset max retries creates infinite retries; we use the ObjectStore default instead
+            max_retries: max_attempts
+                .unwrap_or(NonZeroUsize::new(10).expect("non-zero"))
+                .into(),
+            backoff: BackoffConfig {
+                init_backoff: (*interval).into(),
+                max_backoff: (*interval).into(),
+                base: 1.,
+            },
+            retry_timeout: Into::<std::time::Duration>::into(*interval)
+                * u32::try_from(
+                    max_attempts
+                        .unwrap_or(NonZeroUsize::new(10).expect("non-zero"))
+                        .get(),
+                )
+                .expect("explicit max attempts fits in u32"),
+        },
+        RetryPolicy::Exponential {
+            initial_interval,
+            factor,
+            max_attempts,
+            max_interval,
+        } => RetryConfig {
+            // retry policy with unset max retries creates infinite retries; we use the ObjectStore default instead
```

Comment:
Won't this be confusing to setup an exponential retry intentionally with infinite retry attempts and only end up doing 15 retries ?

Same question for `FixedDelay`